index.js

```diff
@@ -20,6 +22,110 @@ app.get('/info', (req, res) => {
   res.send({ serverName: packageData.name, serverVersion: packageData.version });
 });
 
+// route for displaying all products
+app.get('/products/all', (req, res) => {
+
+  var jsonOutputObject = {
+    products : []
+  };
+
+  for(var i = 0; i < productData.products.length; i++) {
+    
+    jsonOutputObject.products.push(productData.products[i])
+
+    // Getting category name for the product
+    for(var j = 0; j < categoryData.categories.length; j++) {
```

Comment:
Use the [for..of loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) instead.